app/models/user_project_role.rb

```diff
@@ -0,0 +1,15 @@
+class UserProjectRole < ActiveRecord::Base
```

Comment:
I'm not sure you need to use this Join model. 
I think the nicer approach is to use the has_and_belongs_many approach: http://guides.rubyonrails.org/association_basics.html#the-has-and-belongs-to-many-association

That way the User and Project models ```has_and_belongs_to_many :ProjectRole```. And interactions seem a bit more intuitive with user.project_roles and project.project_roles.